src/main/java/net/glowstone/block/blocktype/BlockType.java

```diff
@@ -303,11 +305,28 @@ public final void rightClickBlock(GlowPlayer player, GlowBlock against, BlockFac
             return;
         }
 
-        // TODO: check bounding box
         if (getMaterial().isSolid()) {
-            for (Entity entity : against.getChunk().getEntities()) {
-                if (entity instanceof LivingEntity && entity.getLocation().getBlockX() == target.getX() && entity.getLocation().getBlockY() == target.getY() && entity.getLocation().getBlockZ() == target.getZ()) {
-                    return;
+            GlowChunk[] chunks = new GlowChunk[]{
```

Comment:
`getNearbyEntities()` simplifies the process as well, as you only need to loop through one for-each, and you only need to check if the entities in the list are an instance of `LivingEntity`.